content/code/testing.md

```diff
@@ -66,6 +107,11 @@ Here is an example of such a value being generated.
 
     While randomly generating valid inputs makes sense, randomly generating invalid inputs does not. In the case of invalid inputs, it is much better to have an exhaustive list of all types of cases that are expected to be invalid and always test each one of them.
 
+#### Generic Values Package
+
+When the same generic value need to be used within multiple packages, it can make sense to create a package to store common generic values.
+This package should be located at `./testing/mocks` from the root of the repository, and values should be in a file called `generic.go`.
```

Comment:
Types, values and functions/methods used for testing across the whole project can increase drastically if it is covered with tests.
Seems to me that a single file `generic.go` will be too big for this and it will contain too much different logic.
Since we have a separate sub-package for testing, it makes sense testing entities to be separated in more than one file on multiple levels; for example:
 - helpers.go - file that contains helper testing functions
 - xxx_mocks.go, xxy_mocks.go, xyy_mocks.go, etc - files that contains mocks for the different entities and their methods; can use interface for same data generation methods for example.

Reply:
Currently the Flow DPS is our largest project and it's still in a state where we don't really feel the need to split the generic values between multiple files.

The reason is simple: they are generic values. We do not care how they are generated, we only care about them being random in a deterministic way. Because we do not care about the implementation, it doesn't really make sense to split it into multiple files. In the case of the Flow DPS that would mean having 10 or so files, which we will probably never modify/read (source: https://github.com/optakt/flow-dps/blob/master/testing/mocks/generic.go)

IMO if we ever reach a situation where this part of the guide no longer holds true, we can modify it and deal with it accordingly, but in my personal opinion that will most likely never be the case :)

Reply:
I'm not sure what we should show as an example TBH 🤔 A function that generates one element of a given type deterministically and then another that generates a slice of it, like we have in Flow DPS? I feel like that might be misleading though, because they are implemented this way for the DPS but might be done differently somewhere else, so I'd rather have this part slightly open to interpretation 🤔 WDYT?

Reply:
Ok that makes sense.